onnxruntime/contrib_ops/cpu/murmur_hash3.cc

```diff
@@ -193,14 +197,16 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
       ++output;
     }
   } else {
-    auto input = reinterpret_cast<const uint32_t*>(keys->DataRaw());
-    const auto input_end = input + input_count;
+    auto input = reinterpret_cast<const unsigned char*>(keys->DataRaw());
+    ORT_ENFORCE(input_element_bytes < static_cast<size_t>(std::numeric_limits<int>::max()));
+    int input_num_bytes = static_cast<int>(input_element_bytes);
+    const auto input_end = input + input_count * input_num_bytes;
     while (input != input_end) {
-      MurmurHash3_x86_32(input,
-                         static_cast<int>(input_element_bytes),
+      MurmurHash3_x86_32(reinterpret_cast<const uint32_t*>(input),
```

Comment:
What if `input_num_bytes < sizeof(int)`? Would it access some random bits?

Reply:
I think that e.g if we reinterpret_cast a uint8 to uint32, the uint8 will be padded into a uint32 without access random bits